user-service/src/component-tests/java/uk/gov/gchq/palisade/component/user/service/UserServiceAsyncProxyErrorTest.java

```diff
@@ -0,0 +1,196 @@
+    }
+
+    @AfterEach
+    void tearDown() {
+        cleanCache();
+    }
+
+    @Test
+    void testContextLoads() {
+        assertThat(userServiceAsyncProxy).isNotNull();
+        assertThat(redisTemplate).isNotNull();
+    }
+
+    @Test
+    void testGetUserSuccess() {
+        // Given a user request
+        final UserRequest userRequest = requestFactoryObj.apply(1);
+        final User user = new User().userId("test-user-id");
+        // When adding to the cache
+        this.userServiceAsyncProxy.addUser(user);
+
+        // Then retrieving from the cache
+        final CompletableFuture<AuditableUserResponse> subject = this.userServiceAsyncProxy.getUser(userRequest);
+
+        // Check the CompletableFuture hasn't finished
+        assertThat(subject.isDone()).isFalse();
+        // Then complete the future
+        AuditableUserResponse auditableUserResponse = subject.join();
+        // Then Check it has been completed
+        assertThat(subject.isDone()).isTrue();
+
+        // Then the service suppresses exception and populates Audit object
+        assertThat(auditableUserResponse.getAuditErrorMessage())
+                .as("verify that exception is propagated into an auditable object and returned")
+                .isNull();
+
+        // Then the cached User is the same as the original User
+        assertThat(auditableUserResponse.getUserResponse().getUserId())
+                .isEqualTo(user.getUserId().getId());
+    }
+
+    @Test
+    void testGetUserFailure() {
+        // When adding a user
+        final User user = new User().userId("Not-a-real-user");
+        this.userServiceAsyncProxy.addUser(user);
+
+        // Then retrieving a different user
+        final UserRequest userRequest = requestFactoryObj.apply(1);
+        final CompletableFuture<AuditableUserResponse> subject = this.userServiceAsyncProxy.getUser(userRequest);
+
+        // Check the CompletableFuture hasn't finished
+        assertThat(subject.isDone()).isFalse();
+        // Then complete the future
+        AuditableUserResponse auditableUserResponse = subject.join();
+        // Then Check it has been completed
+        assertThat(subject.isDone()).isTrue();
+
+        // Then check that there is an error message
+        assertThat(auditableUserResponse.getAuditErrorMessage())
+                .as("verify that exception is propagated into an auditable object and returned")
+                .isNotNull();
+
+        // Then check the error message contains the correct message
+        assertThat(auditableUserResponse.getAuditErrorMessage().getError().getMessage())
+                .as("verify that exception is propagated into an auditable object and returned")
+                .isEqualTo(NoSuchUserIdException.class.getName() + ": No userId matching test-user-id found in cache");
+    }
+
+    public static class RedisInitializer implements ApplicationContextInitializer<ConfigurableApplicationContext> {
```

Comment:
This shouldn't need redis. Use caffeine if you need a cache implementation for this test

Reply:
Reworked this class to use Caffeine